src/ui.c

```diff
@@ -85,12 +85,20 @@ int_handler()
 }
 
 void
-put_line(int y, char *str, int len, int so)
+put_line(int y, char *str, int len, int so, int mpos, int mlen)
```

Comment:
Since this function is not only used for putting matcher would it be clearer that this was mure general purpos if `mpos` and `mlen` were renamed `upos` and `ulen` for underline?